tests/spec/w3c/seo-spec.js

```diff
@@ -1,57 +1,93 @@
 "use strict";
 
 import { flushIframes, makeRSDoc, makeStandardOps } from "../SpecHelper.js";
+import { publicationSpaces } from "../../../src/w3c/headers.js";
+import { requiresCanonicalLink } from "../../../src/w3c/seo.js";
 
-function makeTest(uri) {
-  return doc => {
-    const canLink = doc.querySelector("link[rel='canonical']");
-    expect(canLink.href).toBe(uri);
-  };
-}
 describe("W3C - SEO", () => {
   afterAll(flushIframes);
 
   it("defaults to TR as canonical URI", async () => {
-    const test = makeTest("https://www.w3.org/TR/Foo/");
-    test(await makeRSDoc(makeStandardOps()));
+    const doc = await makeRSDoc(makeStandardOps({ specStatus: "REC" }));
+    const href = "https://www.w3.org/TR/Foo/";
+    expect(
+      doc.querySelector(`link[rel='canonical'][href='${href}']`)
+    ).toBeTruthy();
   });
 
   it("sets the canonical URI to TR URI when so configured", async () => {
-    const test = makeTest("https://www.w3.org/TR/Foo/");
-    const ops = makeStandardOps();
-    ops.config.canonicalURI = "TR";
-    test(await makeRSDoc(ops));
+    const href = "https://www.w3.org/TR/Foo/";
+    const ops = makeStandardOps({ canonicalURI: "TR" });
+    const doc = await makeRSDoc(ops);
+    expect(
+      doc.querySelector(`link[rel='canonical'][href='${href}']`)
+    ).toBeTruthy();
```

Comment:
Side note: Might be nice to have a custom jasmine matcher in future. Like:
```js
expect(`link[rel='canonical'][href='${href}']`).toBeInDocument(doc);
```

Reply:
oh! fancy! Didn't know we could do that.

Reply:
Can't yet, but can [extend jasmine](https://jasmine.github.io/tutorials/custom_matcher).